### PR TITLE
MAINT: make galaxy use secrets

### PR DESCRIPTION
**Pull request by anish-mudaraddi**
Galaxy application now utilises sops secrets to set admin emails and IAM credentials
Added Docs for Galaxy deployment
Also I removed manila csi application as we're not using it

### CODE REVIEW

File: charts/dev/materials-galaxy/secrets-templates.yaml/materials-galaxy.yaml
Comment (RJaliliSTFC) [inline charts/dev/materials-galaxy/secrets-templates.yaml/materials-galaxy.yaml:1]: Do you want this directory to be called secrets-templates.yaml?

File: charts/dev/materials-galaxy/secrets-templates.yaml/materials-galaxy.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    clientID: iris-iam-client-id
+    clientSecret: iris-iam-secret
+    # Create a new secret with the following command
+    # openssl rand -base64 32 | head -c 32 | base64
+    cookieSecret: "XXXXXXXXXXXXXXXX"
+  
+  # if you want to setup galaxy where only specific emails have access
+  # USE EITHER THIS OR RESTRICT BY OIDC GROUP 
+  authenticatedEmailsFile:
+    enabled: true
```
Comment (RJaliliSTFC) [inline charts/dev/materials-galaxy/secrets-templates.yaml/materials-galaxy.yaml:14]: if this is enabled true by default, will it error if no file is provided?

File: clusters/dev/galaxy/materials-galaxy-values.yaml [hunk 2]
```diff
@@ -1,16 +1,11 @@
 oauth2-proxy:
+
+
   extraArgs:
     # limit access to stfc-cloud-dev group
```
Comment (RJaliliSTFC) [inline clusters/dev/galaxy/materials-galaxy-values.yaml:5]: Comment is old now?

**Review by RJaliliSTFC — CHANGES_REQUESTED** [review]

File: charts/dev/materials-galaxy/secrets-templates.yaml/materials-galaxy.yaml
Comment (anish-mudaraddi) [reply charts/dev/materials-galaxy/secrets-templates.yaml/materials-galaxy.yaml:1]: any better names you can think of? Its just to store "an example secret" which you can copy-paste into an encrypted sops file and change the values

File: charts/dev/materials-galaxy/secrets-templates.yaml/materials-galaxy.yaml (hunk 1 above)
Comment (anish-mudaraddi) [reply charts/dev/materials-galaxy/secrets-templates.yaml/materials-galaxy.yaml:14]: This is just a template - so its not being used - but fair point I'll write a comment to set this to false it if you're restricting by OIDC group

File: charts/dev/materials-galaxy/secrets-templates.yaml/materials-galaxy.yaml
Comment (RJaliliSTFC) [reply charts/dev/materials-galaxy/secrets-templates.yaml/materials-galaxy.yaml:1]: Comment was mostly for the .yaml part, think it'd be more clear as just "secrets-templates"

Comment (anish-mudaraddi) [reply charts/dev/materials-galaxy/secrets-templates.yaml/materials-galaxy.yaml:1]: oh yea - typo :sweat_smile:

**Review by RJaliliSTFC — APPROVED** [review]

**Pull request merged**